docs/user_guide/model_management.md

```diff
@@ -224,6 +224,9 @@ request is received under
 'config.pbtxt' is modified in place. The swap file is not part of the model
 configuration, so its presence in the model directory may be detected as a new file
 and cause the model to fully reload when only an update is expected.
+  * When a model configuration is reloaded existing model instances will be
+checked against the new configuration and be reused if possible. If a model
+instance is reused it will retain its original name.
```

Comment:
> If a model instance is reused it will retain its original name.
>
>> Is this true? IIRC the current behavior involved popping instances from the back in some reverse order and just renaming them. And current core PR just normalizes the names to always include the index.
>>
>>> I should have described it more clearly. The instance cannot be renamed after it is created, but the selection algorithm for an instance to reuse does not honor the name of the instance that it is picking. For example:
>>>
>>> Existing instances:   model_1_0   model_1_1   model_1_2
The user reduced the instance group count from 3 to 2, ideally, model_1_0 and model_1_1 should be instances remaining after the update. Since all instances are no different to each other beside their names, the algorithm may pick any two existing instance to be reused. The current behavior will pick model_1_2 and model_1_1 as the requested (or ideal) instance model_1_0 and model_1_1, which:
>>> 
>>> New instances:   model_1_2   model_1_1
It looks like the instance model_1_0 is renamed to model_1_2, but model_1_2 is actually picked instead of ideally picking model_1_0 at its place. (Note: nothing is renamed above, only which instance is picked and where it is placed on the list).

>>> This is (ok?), if the instances are named by default. If the user would to flatten a instance group, for example:
>>>{
>>>  {kind: CPU, count: 1, name: "my_name_1"},
>>>  {kind: CPU, count: 1, name: "my_name_2"},
>>>  {kind: CPU, count: 1, name: "my_name_3"}
>>>}
>>> and then remove my_name_3:
>>>
>>>{
>>>  {kind: CPU, count: 1, name: "my_name_1"},
>>>  {kind: CPU, count: 1, name: "my_name_2"}
>>>}
>>>
>>> Since the instance group name is not honored during reuse, the user will see my_name_3   my_name_2 instead of the expected my_name_1   my_name_2. (Note: nothing is renamed, but my_name_2 and my_name_3 are picked for the reuse, instead of my_name_1 and my_name_2)
>>>
>>> Then, a solution is to add the name into the instance signature, so the group with the correct name will be picked. This works perfectly with user specified names, but not with default names. Maybe we should include names in instance matching if the name is specified by the user. If the name is not specified by the user, we should match with other equivalent instances that do not have a name specified by the user. In this case, maybe we can simply stop filling-in default names and leave those unspecified names empty. This naturally distinguished those that have names specified or not (!name.empty() vs name.empty()), so the instance matching algorithm can simply take the name into account without any worries. At the end, a name will be requested (i.e. LOG_ERROR << instance.Name()). If the name is not empty, the name is returned, If the name is empty, a default name can be assembled immediately.
>>>
>>> Do we want to implement the solution on this ticket? Or we can add note on the limitation, and enhance it on a different ticket?

Moving comments so far to thread, my bad.

Reply:
Personally, I think this is not important to document right now. If we're planning some reworks or reconsiderations of naming behavior as discussed in [core PR](https://github.com/triton-inference-server/core/pull/231), I think we can just close this PR and leave out this comment for now and update the documentation when we make some behavioral changes. 

What do you think @GuanLuo @kthui ?

Reply:
Use hyperlink... I am leaning towards that this documentation is not necessary as well: what additional information that the user will gain from reading it? When the instance is "reused" (I am not sure about the use of this word as well), is the reader expecting the instance name will be different and thus we now need to state that the name is the same?

Reply:
I agree the documentation does not help the user reading it. I will close this PR.